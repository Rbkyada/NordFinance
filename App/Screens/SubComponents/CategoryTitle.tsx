import React, { memo } from 'react';
import { StyleProp, TextStyle } from 'react-native';
import LightTheme from '@Theme/LightTheme';
import { CustomText } from '@CommonComponent/index';
import { fonts } from '@Utils/Constant';

interface CategoryTitleProps {
  title: string;
  textStyle?: StyleProp<TextStyle>;
}

const CategoryTitle = memo((props: CategoryTitleProps) => {
  const { title, textStyle } = props;

  return (
    <CustomText
      xlarge
      style={[
        fonts.Bold,
        { color: LightTheme.themeColor },
        textStyle && textStyle,
      ]}>
      {title}
    </CustomText>
  );
});

export { CategoryTitle };
