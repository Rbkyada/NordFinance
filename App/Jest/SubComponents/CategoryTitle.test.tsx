import renderer from 'react-test-renderer';
import { CategoryTitle } from '@SubComponents/CategoryTitle';

describe('CategoryTitle component', () => {
  it('create Snapshot', () => {
    const title = 'Category Title';
    const textStyle = { color: 'red' };
    const tree = renderer
      .create(<CategoryTitle title={title} textStyle={textStyle} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
