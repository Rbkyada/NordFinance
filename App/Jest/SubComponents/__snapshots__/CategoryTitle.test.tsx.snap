// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTitle component create Snapshot 1`] = `
<Text
  numberOfLines={0}
  style={
    [
      {
        "fontSize": 18,
      },
      {
        "color": "#1c1c1e",
      },
      {
        "fontFamily": "DMSans-Regular",
      },
      [
        {
          "fontFamily": "DMSans-Bold",
        },
        {
          "color": "rgba(158, 172, 219, 1)",
        },
        {
          "color": "red",
        },
      ],
    ]
  }
  xlarge={true}
>
  Category Title
</Text>
`;
